tests/Unit/Evolution/Systems/Cce/Test_KleinGordonWorldtubeData.cpp

```diff
@@ -0,0 +1,362 @@
+
+  const auto& time_buffer = buffer_updater.get_time_buffer();
+  for (size_t i = 0; i < time_buffer.size(); ++i) {
+    CHECK(time_buffer[i] == approx(target_time - 0.1 + 0.01 * i));
+  }
+  CHECK(buffer_updater.get_extraction_radius() == 100.0);
+
+  // Test the `pup` function of `KleinGordonWorldtubeH5BufferUpdater`.
+  // Serialize and deserialize `buffer_updater` and repeat the checks above.
+  auto serialized_and_deserialized_updater =
+      serialize_and_deserialize(buffer_updater);
+
+  Variables<klein_gordon_input_tags> coefficients_buffers_from_serialized{
+      (buffer_size + 2 * interpolator_length) * square(computation_l_max + 1)};
+  size_t time_span_start_from_serialized = 0;
+  size_t time_span_end_from_serialized = 0;
+  serialized_and_deserialized_updater.update_buffers_for_time(
+      make_not_null(&coefficients_buffers_from_serialized),
+      make_not_null(&time_span_start_from_serialized),
+      make_not_null(&time_span_end_from_serialized), target_time,
+      computation_l_max, interpolator_length, buffer_size);
+
+  const auto& time_buffer_from_serialized =
+      serialized_and_deserialized_updater.get_time_buffer();
+  for (size_t i = 0; i < time_buffer.size(); ++i) {
+    CHECK(time_buffer_from_serialized[i] ==
+          approx(target_time - 0.1 + 0.01 * i));
+  }
+  CHECK(serialized_and_deserialized_updater.get_extraction_radius() == 100.0);
+
+  // Compare the modal data of `buffer_updater` and
+  // `serialized_and_deserialized_updater`, which goes through the write-read
+  // process, with what we generated earilier. The fiducial modal data are
+  // stored in an object of `KleinGordonDummyBufferUpdater`.
+  time_span_start = 0;
+  time_span_end = 0;
+  const KleinGordonDummyBufferUpdater dummy_buffer_updater{
+      time_buffer, extraction_radius, amplitude, frequency, computation_l_max};
+  Variables<klein_gordon_input_tags> expected_coefficients_buffers{
+      (buffer_size + 2 * interpolator_length) * square(computation_l_max + 1)};
+  dummy_buffer_updater.update_buffers_for_time(
+      make_not_null(&expected_coefficients_buffers),
+      make_not_null(&time_span_start), make_not_null(&time_span_end),
+      target_time, computation_l_max, interpolator_length, buffer_size);
+
+  Approx modal_approx =
+      Approx::custom()
+          .epsilon(std::numeric_limits<double>::epsilon() * 1.0e5)
+          .scale(1.0);
+
+  tmpl::for_each<klein_gordon_input_tags>(
+      [&expected_coefficients_buffers, &coefficients_buffers_from_file,
+       &coefficients_buffers_from_serialized, &modal_approx](auto tag_v) {
+        using tag = typename decltype(tag_v)::type;
+        INFO(db::tag_name<tag>());
+        const auto& test_lhs = get<tag>(expected_coefficients_buffers);
+        const auto& test_rhs = get<tag>(coefficients_buffers_from_file);
+        CHECK_ITERABLE_CUSTOM_APPROX(test_lhs, test_rhs, modal_approx);
+        const auto& test_rhs_from_serialized =
```

Comment:
I know this was just copied from `Test_WorldtubeData.cpp`, but can you rename these to
- `test_lhs` -> `expected_coefs`
- `test_rhs` -> `kg_coefs`
- `test_rhs_from_serialized` -> `serialized_kg_coefs`

so the names are clearer?